elastic/integrations
# microsoft_sqlserver/data_stream/transaction_log: Add support for `fetch_from_all_databases` for mssql

## What does this PR do?

To make the Microsoft SQL Server integration setup easier, there's no simple way of getting the predefined queries run against all databases of a server to get the desired metrics. Currently, the only way to add, modify or remove databases (both system and user-defined) is to do them manually while configuring the integration. However, if there are a large number of databases on the server, this manual process can be tedious and time-consuming. To address this issue, there is an option to enable fetching the required `transaction_log` metrics from all databases on the server, including system and user-defined databases.

This option is controlled by a boolean field called `fetch_from_all_databases`. By default, this feature is set to `false`. To use it, simply set the field to `true`. This will allow the integration to automatically find and fetch metrics from all databases on the server, without requiring users to manually enter each individual database.

Also, fixed some issues in the documentation, etc.

## Checklist


## How to test this PR locally

- Tested on Mac OS Ventura 13.3.1
- Tested against mcr.microsoft.com/azure-sql-edge
- Add a user-defined database called `hello_world`. Already four system databases are present: `master`, `model`, `tempdb` and `msdb`
- Metrics should now be available for all 5 databases without the need to specify the database names in the configuration manually

```
docker run --cap-add SYS_PTRACE -e 'ACCEPT_EULA=Y' -e 'MSSQL_SA_PASSWORD=<password>' -p 1433:1433 --name sqledge -d mcr.microsoft.com/azure-sql-edge
```

User: `SA`
Password: *****

## Related issues

- Closes https://github.com/elastic/integrations/issues/4108
- Closes https://github.com/elastic/integrations/issues/7499

## Screenshots

<img width="1714" alt="screenshot-kbn-dashboard" src="https://github.com/elastic/integrations/assets/51480165/653a6609-831a-4051-92d5-6ccf101b3328">
